common/src/main/scala/net/psforever/packet/game/SquadWaypointRequest.scala

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2019 PSForever
+package net.psforever.packet.game
+
+import net.psforever.packet.{GamePacketOpcode, Marshallable, PacketHelpers, PlanetSideGamePacket}
+import net.psforever.types.Vector3
+import scodec.{Attempt, Codec, Err}
+import scodec.codecs._
+import shapeless.{::, HNil}
+
+/**
+  * Actions that can be requested of the specific waypoint.
+  */
+object WaypointEventAction extends Enumeration {
+  type Type = Value
+
+  val
+  Add,
+  Unknown1,
+  Remove,
+  Unknown3 //unconfirmed
+  = Value
+
+  implicit val codec : Codec[WaypointEventAction.Value] = PacketHelpers.createEnumerationCodec(enum = this, uint2)
+}
+
+/**
+  * na
+  * @param zone_number the zone
+  * @param pos the continental map coordinate location of the waypoint;
+  *            the z-coordinate is almost always 0.0
+  */
+final case class WaypointInfo(zone_number : Int,
+                              pos : Vector3)
+
+/**
+  * na
+  * @param request_type the action to be performed
+  * @param char_id the unique id of player setting the waypoint
+  * @param waypoint_type the waypoint being updated;
+  *                      0-3 for the standard squad waypoints numbered "1-4";
+  *                      4 for the squad leader experience waypoint;
+  *                      cycles through 0-3 continuously
+  *
+  * @param unk4 na
+  * @param waypoint_info essential data about the waypoint
+  */
+final case class SquadWaypointRequest(request_type : WaypointEventAction.Value,
+                                      char_id : Long,
```

Comment:
We should also consider creating a `type CharacterID = Long` typedef somewhere to improve type checking/readability